### PR TITLE
Add sample configs and TF configs for the fleet tenancy integration use case

### PR DESCRIPTION
**Pull request by nan-yu**

### CODE REVIEW

File: fleet-tenancy/README.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+- After the step, you need to create and download the service account’s private
+  key JSON file.
+
+## 2. Configure Fleet Default Member Config ConfigManagement
+
+This step configures the Fleet Default Member Config ConfigManagement.
+
+Terraform configs: [link](terraform/2-fleet-default-config).
+
+Apply the resources via Terraform: [README.md](terraform/2-fleet-default-config/README.md).
```
[the remark this answers is not in the document]
Comment (janetkuo) [reply fleet-tenancy/README.md:57]: Should we link to Namespace Selector doc and explain a bit about how it works (i.e. select the fleet namespaces of a scope by selecting specific labels), too?

File: fleet-tenancy/config/scope-backend/namespace-selector.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+# limitations under the License.
+apiVersion: configmanagement.gke.io/v1
+kind: NamespaceSelector
+metadata:
+  name: backend-scope
+spec:
+  mode: dynamic
+  selector:
+    matchLabels:
+      fleet.gke.io/fleet-scope: backend
```
Comment (janetkuo) [inline fleet-tenancy/config/scope-backend/namespace-selector.yaml:22]: We could add a comment here, e.g.
```
# Change the value of this label selector to match different scopes by scope name 
```

File: fleet-tenancy/config/scope-backend/networking.k8s.io_v1_networkpolicy_deny-all.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: deny-all
+  annotations:
+    configmanagement.gke.io/namespace-selector: backend-scope
```
Comment (janetkuo) [inline fleet-tenancy/config/scope-backend/networking.k8s.io_v1_networkpolicy_deny-all.yaml:19]: We could add a comment here, something like:
```
# Actuate this resource in the namespaces with labels matched by the namespace selector
```

File: fleet-tenancy/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (nan-yu) [reply fleet-tenancy/README.md:69]: Added more explanation.

File: fleet-tenancy/config/scope-backend/namespace-selector.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# Copyright 2023 Google LLC
```
[the remark this answers is not in the document]
Comment (nan-yu) [reply fleet-tenancy/config/scope-backend/namespace-selector.yaml]: updated

[the remark this answers is not in the document]
Comment (nan-yu) [reply fleet-tenancy/config/scope-backend/namespace-selector.yaml]: done

File: fleet-tenancy/config/scope-backend/namespace-selector.yaml (hunk 2 above)
Comment (nan-yu) [reply fleet-tenancy/config/scope-backend/namespace-selector.yaml]: added

File: fleet-tenancy/config/scope-backend/networking.k8s.io_v1_networkpolicy_deny-all.yaml (hunk 3 above)
Comment (nan-yu) [reply fleet-tenancy/config/scope-backend/networking.k8s.io_v1_networkpolicy_deny-all.yaml]: added

File: fleet-tenancy/config/scope-frontend/resource-quota-pvc.yaml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ResourceQuota Reference: https://kubernetes.io/docs/concepts/policy/resource-quotas/
+kind: ResourceQuota
+apiVersion: v1
+metadata:
+  name: pvc
```
[the remark this answers is not in the document]
Comment (nan-yu) [reply fleet-tenancy/config/scope-frontend/resource-quota-pvc.yaml]: done

File: fleet-tenancy/terraform/1-fleet/README.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
+# Set up your fleet
+
+The Terraform configurations in this directory sets up a fleet, along with the a list of enabled API services, and a service account that Terraform can use to access the Google Cloud APIs.
+
+## Usage
+
+See the [variable definitions file] for an exhaustive list of variables.
+These can be provided as [command line variables] at runtime.
```
[the remark this answers is not in the document]
Comment (nan-yu) [reply fleet-tenancy/terraform/1-fleet/README.md:8]: added

File: fleet-tenancy/README.md [hunk 7]
```diff
@@ -0,0 +1,167 @@
+# Configure Fleet Tenancy Resources with Config Sync
+
+[Fleet team management] provides a way to configure Namespaces and RBAC
+resources at the scope level.
+[Config Sync] extends this capability, allowing for the configuration of
+additional namespace-scoped resources at the scope level.
+
+Here is an example of a Fleet with two scopes: `backend` and `frontend`.
```
Comment (janetkuo) [inline fleet-tenancy/README.md:8]: To make it a bit more explicit following the Fleet team doc: 

```suggestion
Here is an example of a Fleet with two team scopes: `backend` and `frontend`.
```

File: fleet-tenancy/README.md [hunk 8]
```diff
@@ -0,0 +1,167 @@
+# Configure Fleet Tenancy Resources with Config Sync
```
Comment (janetkuo) [inline fleet-tenancy/README.md:1]: How about 

```suggestion
# Manage Resources of Teams and Fleet Namespaces with Config Sync
```

File: fleet-tenancy/README.md [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+
+## Cleanup
+
+- Delete Fleet Scopes and Namespaces
+- Unregister and delete clusters
+- Delete Fleet default member config
+- Delete the service account and disable services
+
+To clean up, you can iterate through each terraform config directory in the
+reverse order and run the terraform destroy command with your GCP project and
```
Comment (janetkuo) [inline fleet-tenancy/README.md:159]:
```suggestion
reverse order and run the `terraform destroy` command with your GCP project and
```

File: fleet-tenancy/README.md [hunk 10]
```diff
@@ -0,0 +1,167 @@
...
+Apply the resources via Terraform: [README.md](terraform/4-scopes-and-namespaces/README.md).
+
+## 5. Sync more resources to the scope
+
+All scopes share the same source of truth. If you want to sync more sources, you
+can add the configurations to the source repository with the correct
+NamespaceSelector annotation. Config Sync will automatically reconcile those
+resources.
+
+## 6. Remove a cluster from a scope
```
Comment (janetkuo) [inline fleet-tenancy/README.md:102]: I'd suggest removing step 6 and 7 to keep it concise. Instead, have a paragraph that says the resources will be synced dynamically to fleet namespaces in team scopes, when new clusters is added to or removed from a scope, or when a fleet namespace is added or removed.

File: fleet-tenancy/README.md [hunk 11]
```diff
@@ -0,0 +1,167 @@
...
+This step includes the following operations:
+- Create team scopes: `backend` and `frontend`
+- Add clusters to scopes
+- Create fleet Namespaces
+
+Terraform configs: [link](terraform/4-scopes-and-namespaces).
+
+Apply the resources via Terraform: [README.md](terraform/4-scopes-and-namespaces/README.md).
+
+## 5. Sync more resources to the scope
```
Comment (janetkuo) [inline fleet-tenancy/README.md:95]:
```suggestion
```

Move the diagram below that shows the UI of resources being created in the fleet namespaces, and talk about what we've accomplished so far in step 4. And then mention that users can sync more resources by adding configs in the repo.

**Review by janetkuo — COMMENTED** [review]
I think we could try to make this more concise by combining the terraform configs so that it doesn't need to be applied multiple times, but we can iterate on that later. 

Mostly looks good and just some minor comments.

File: fleet-tenancy/README.md (hunk 11 above)
Comment (nan-yu) [reply fleet-tenancy/README.md]: done

File: fleet-tenancy/README.md (hunk 10 above)
Comment (nan-yu) [reply fleet-tenancy/README.md]: done

**Review by janetkuo — APPROVED** [review]

**Pull request merged**